src/eos/adiabatic_glmmhd.cpp

```diff
@@ -37,88 +37,19 @@ void AdiabaticGLMMHDEOS::ConservedToPrimitive(MeshData<Real> *md) const {
   auto jb = md->GetBlockData(0)->GetBoundsJ(IndexDomain::entire);
   auto kb = md->GetBlockData(0)->GetBoundsK(IndexDomain::entire);
 
-  auto gam = GetGamma();
-  auto gm1 = gam - 1.0;
-  auto density_floor_ = GetDensityFloor();
-  auto pressure_floor_ = GetPressureFloor();
-  auto e_floor_ = GetInternalEFloor();
-
   auto pkg = md->GetBlockData(0)->GetBlockPointer()->packages.Get("Hydro");
   const auto nhydro = pkg->Param<int>("nhydro");
   const auto nscalars = pkg->Param<int>("nscalars");
 
+  auto this_on_device = (*this);
```

Comment:
This is required to get access to the `GetXXX` calls, doesn't it?
Have you profiled how much overhead is introduced by those calls?
Previously, these were captured scalars. Now these functions are called for each index.